src/workers/jobs.js

```diff
@@ -521,37 +521,57 @@ export async function exportCampaign(job) {
   const messageCsv = Papa.unparse(finalCampaignMessages)
 
   if (process.env.AWS_ACCESS_AVAILABLE || (process.env.AWS_ACCESS_KEY_ID && process.env.AWS_SECRET_ACCESS_KEY)) {
-    const s3bucket = new AWS.S3({ params: { Bucket: process.env.AWS_S3_BUCKET_NAME } })
-    const campaignTitle = campaign.title.replace(/ /g, '_').replace(/\//g, '_')
-    const key = `${campaignTitle}-${moment().format('YYYY-MM-DD-HH-mm-ss')}.csv`
-    const messageKey = `${key}-messages.csv`
-    let params = { Key: key, Body: campaignCsv }
-    await s3bucket.putObject(params).promise()
-    params = { Key: key, Expires: 86400 }
-    const campaignExportUrl = await s3bucket.getSignedUrl('getObject', params)
-    params = { Key: messageKey, Body: messageCsv }
-    await s3bucket.putObject(params).promise()
-    params = { Key: messageKey, Expires: 86400 }
-    const campaignMessagesExportUrl = await s3bucket.getSignedUrl('getObject', params)
-    await sendEmail({
-      to: user.email,
-      subject: `Export ready for ${campaign.title}`,
-      text: `Your Spoke exports are ready! These URLs will be valid for 24 hours.
-
-      Campaign export: ${campaignExportUrl}
-
-      Message export: ${campaignMessagesExportUrl}`
-    })
-    log.info(`Successfully exported ${id}`)
+    try {
+      const s3bucket = new AWS.S3({ params: { Bucket: process.env.AWS_S3_BUCKET_NAME } })
+      const campaignTitle = campaign.title.replace(/ /g, '_').replace(/\//g, '_')
+      const key = `${campaignTitle}-${moment().format('YYYY-MM-DD-HH-mm-ss')}.csv`
+      const messageKey = `${key}-messages.csv`
+      let params = { Key: key, Body: campaignCsv }
+      await s3bucket.putObject(params).promise()
+      params = { Key: key, Expires: 86400 }
+      const campaignExportUrl = await s3bucket.getSignedUrl('getObject', params)
+      params = { Key: messageKey, Body: messageCsv }
+      await s3bucket.putObject(params).promise()
+      params = { Key: messageKey, Expires: 86400 }
+      const campaignMessagesExportUrl = await s3bucket.getSignedUrl('getObject', params)
```

Comment:
in case email gets borked, and since you're changing this code, can you drop a `log.info` log here with the `campaignMessagesExportUrl` -- that way, if the email fails, at least the heroku admin can see the url from the logs.

Reply:
No problem, I think directly catching errors on the email and logging there should do the trick since the try/catch would catch an error at any step and the URLs may not be created (or in scope unless defined before the block). The alternative would be using `.catch` on every `await` function, I guess.

See: https://github.com/MoveOnOrg/Spoke/pull/575/files#diff-ee93a980f871b21055cf3e181d6583beR543